### PR TITLE
enable publish async

### PR DESCRIPTION
**Pull request by Liuhaai**

### CODE REVIEW

File: src/w3bstream_client_python/client.py [hunk 1]
```diff
@@ -42,60 +41,61 @@ def publish_event_sync(self, header: Header, payload: bytes) -> requests.Respons
...
+            for header, payload in self._fetch_event_batch():
+                events.append({
+                    'device_id': header.device_id,
+                    'event_type': header.event_type,
+                    'payload': payload.decode('utf-8'),
+                    'timestamp':  int(round(header.timestamp.timestamp())),
+                })
+            # publish events
+            resp = self._publish_event(events)
+            if resp.status_code >= 400:
```
Comment (Liuhaai) [inline src/w3bstream_client_python/client.py:75]: modified

File: src/w3bstream_client_python/client.py [hunk 2]
```diff
@@ -17,18 +17,17 @@ def __init__(self, device_id: str, event_type: str = 'DEFAULT', timestamp: datet
 
 
 PUBLISH_INTERVAL = 5
-PUBLISH_BATCH_SIZE = 10
+PUBLISH_BATCH_SIZE = 1
```
Comment (Liuhaai) [inline src/w3bstream_client_python/client.py]: modified

File: src/w3bstream_client_python/client.py [hunk 3]
```diff
@@ -42,60 +41,61 @@ def publish_event_sync(self, header: Header, payload: bytes) -> requests.Respons
...
+                    'event_type': header.event_type,
+                    'payload': payload.decode('utf-8'),
+                    'timestamp':  int(round(header.timestamp.timestamp())),
+                })
+            # publish events
+            resp = self._publish_event(events)
+            if resp.status_code >= 400:
+                # TODO: Support retry mechanism when failed to publish
+                sys.stderr.write(
+                    "An error occurred when publishing the data to W3bstream: status_code %d, body: %s" % (resp.status_code, resp.text))
```
Comment (Liuhaai) [inline src/w3bstream_client_python/client.py:78]: modified

File: src/w3bstream_client_python/client.py (hunk 2 above)
Comment (envestcc) [reply src/w3bstream_client_python/client.py]: Currently, publish async only works well if `PUBLISH_BATCH_SIZE = 1`.  It is recommended to rename it as private, such as `_PUBLISH_BATCH_SIZE`, indicating that it should not be modified from outside the module.

**Review by envestcc — APPROVED** [review]

**Review by envestcc — APPROVED** [review]
also need increase module version

**Pull request merged**